File: src/main/java/kr/co/lunatalk/domain/exhibition/domain/ExhibitionProduct.java
package kr.co.lunatalk.domain.exhibition.domain;

import jakarta.persistence.*;
import kr.co.lunatalk.domain.common.domain.BaseTimeEntity;
import kr.co.lunatalk.domain.product.domain.Product;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ExhibitionProduct extends BaseTimeEntity {

	@Id @GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "exhibition_id")
	private Exhibition exhibition;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "product_id")
	private Product product;

	private int sortOrder;

	@Builder
	public ExhibitionProduct(Exhibition exhibition, Product product, int sortOrder) {
		this.exhibition = exhibition;
		this.product = product;
		this.sortOrder = sortOrder;
	}

	public static ExhibitionProduct createExhibitionProduct(Exhibition exhibition, Product product, int sortOrder) {
		return ExhibitionProduct.builder()
			.exhibition(exhibition)
			.product(product)
			.sortOrder(sortOrder)
			.build();
	}
}

File: lunatalk-core/src/main/kotlin/kr/co/lunatalk/domain/exhibition/domain/ExhibitionProduct.kt
package kr.co.lunatalk.domain.exhibition.domain

import jakarta.persistence.*
import kr.co.lunatalk.domain.common.domain.BaseTimeEntity
import kr.co.lunatalk.domain.product.domain.Product

@Entity
open class ExhibitionProduct protected constructor() : BaseTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exhibition_id")
    open var exhibition: Exhibition? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    open var product: Product? = null
        protected set

    open var sortOrder: Int = 0
        protected set

    constructor(exhibition: Exhibition, product: Product, sortOrder: Int) : this() {
        this.exhibition = exhibition
        this.product = product
        this.sortOrder = sortOrder
    }

    companion object {
        fun createExhibitionProduct(exhibition: Exhibition, product: Product, sortOrder: Int): ExhibitionProduct {
            return ExhibitionProduct(exhibition, product, sortOrder)
        }
    }
}
